src/blant-utils.c

```diff
@@ -175,10 +180,14 @@ void LoadMagicTable(void)
 
 // You provide a permutation array, we fill it with the permutation extracted from the compressed Permutation mapping.
 // There is the inverse transformation, called "EncodePerm", in createBinData.c.
-void ExtractPerm(unsigned char perm[_k], int i)
-{
+void ExtractPerm(unsigned char perm[_k], Gint_type Gint)
+{   
+    if(_k > 8) {
+	    smaller_canon_map(Gint, _k, perm);
```

Comment:
It looks like smaller_canon_map will be called twice during decomposition, once in L_K and again here. That's too expensive, try to make it get called only once per sampled graphlet.

Reply:
That's why I was using smaller_canon_map directly instead of ExtractPerm

I'm not sure how to do it properly

Reply:
Made necessary improvement to optimize it